pkg/build/api/types.go

```diff
@@ -475,6 +475,10 @@ const (
 	// build pod but one was already present.
 	StatusReasonBuildPodExists StatusReason = "BuildPodExists"
 
+	// StatusReasonInvalidEnvironmentVariable indicates that an error occurred processing
+	// the supplied options for environment variables in the build strategy environment
+	StatusReasonInvalidEnvironmentVariable StatusReason = "InvalidEnvironmentVariable"
```

Comment:
@jim-minter are you suggesting a name more like invalidenvironmentvariablereference or invalidenvironmentvariablefrom or something?

For my part i'm ok with this as is, but maybe an easy improvement would be "UnresolveableEnvironmentVariable"?